extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiConfig.java

```diff
@@ -0,0 +1,115 @@
+package io.quarkus.swaggerui.deployment;
+
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+/**
+ * Please @see <a href="https://swagger.io/docs/open-source-tools/swagger-ui/usage/configuration"</a>
+ */
+@ConfigRoot
+class SwaggerUiConfig {
+    /**
+     * The path where Swagger UI is available.
+     * <p>
+     * The value `/` is not allowed as it blocks the application from serving anything else.
+     */
+    @ConfigItem(defaultValue = "/swagger-ui")
+    String path;
+
+    /**
+     * If this should be included every time. By default, this is only included when the application is running
+     * in dev mode.
+     */
+    @ConfigItem(defaultValue = "false")
+    boolean alwaysInclude;
+
+    /**
+     * URL to fetch external configuration document from.
+     */
+    @ConfigItem
+    Optional<String> configUrl;
+
+    /**
+     * Enables deep linking for tags and operations.
+     * 
+     * @see <a href="https://swagger.io/docs/open-source-tools/swagger-ui/usage/deep-linking</a>
+     */
+    @ConfigItem
+    Optional<Boolean> deepLinking;
+
+    /**
+     * The default expansion depth for models (set to -1 completely hide the models).
+     */
+    @ConfigItem
+    OptionalInt defaultModelsExpandDepth;
+
+    /**
+     * The default expansion depth for the model on the model-example section.
+     */
+    @ConfigItem
+    OptionalInt defaultModelExpandDepth;
```

Comment:
I find it a bit confusing to have an option with an s and one without. It's like that in Swagger UI?

Reply:
Me too, i just copied the original configuration properties names.